pkg/controller/provider/controller.go

```diff
@@ -196,11 +198,46 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 			deployment)
 
 		// If the deployment does not exist
-		if k8serr.IsNotFound(err) {
-			r.CreateOVAServerDeployment(provider, ctx)
-		} else if err != nil {
+		if err != nil {
+			if k8serr.IsNotFound(err) {
+				r.CreateOVAServerDeployment(provider, ctx)
+				provider.Status.Phase = initializing
+				provider.Status.SetCondition(
+					libcnd.Condition{
+						Type:     initializing,
+						Status:   True,
+						Category: Required,
+						Message:  "The OVA server being inizialized.",
+					})
+				err = r.Status().Update(context.TODO(), provider.DeepCopy())
+				if err != nil {
+					return
+				}
+				result.RequeueAfter = 5 * time.Second
+				return
+			}
 			return
 		}
+
+		// The ova server pod is not running yet
+		if deployment.Status.AvailableReplicas == 0 {
+			if time.Since(provider.CreationTimestamp.Time).Minutes() <= OvaTimeout {
```

Comment:
umm i don't think its matter if the time is measured from the provider creation or the deployment, if the deployment failing for any reason the pod check should cover it (its simply wont run). but really I didn't address the case the deployment isnt creating at all, I"ll add this check as well.

Reply:
so `CreateOVAServerDeployment` functoin already handling errors, so I simply returned it to the controller and added the error msg to the condition as well, I think it should cover all cases now.

![Screenshot from 2023-12-06 13-44-14](https://github.com/kubev2v/forklift/assets/73168733/aa640265-8b4f-4af1-9ea1-5647e72ae275)

Reply:
> so `CreateOVAServerDeployment` functoin already handling errors, so I simply returned it to the controller and added the error msg to the condition as well, I think it should cover all cases now.
> 
> ![Screenshot from 2023-12-06 13-44-14](https://private-user-images.githubusercontent.com/73168733/288393376-aa640265-8b4f-4af1-9ea1-5647e72ae275.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIxOTg4NjIsIm5iZiI6MTcwMjE5ODU2MiwicGF0aCI6Ii83MzE2ODczMy8yODgzOTMzNzYtYWE2NDAyNjUtOGI0Zi00YWYxLTllYTEtNTY0N2U3MmFlMjc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEwVDA4NTYwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1MWNjMDA3NTZlMTZjMTBkYzg4MGMyMDhhNDBkZGQ5MGFjYmMxNDNiZGQ1MzgyNmMzNzY2ZDdjNmUwY2M4YWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YP9blVjKSVA6R8G4q6OKDRjSqJwKq0s1SRLwWi4UkjM)

that this particular issue can't happen now because we are supposed to create the provider only when it doesn't exist, no?

Reply:
It can be since the server creation depends on several steps like the PV, PVC creation and then the deployment + service, in that case for any error in the creation we will change the status immediately to failed, without retries. if you referring to the specific PV issue, its shouldn't happen (it currently does because we have a bug which i'm currently working on - [the clean up issue](https://github.com/kubev2v/forklift/issues/672)) just added it as an example to show this case.

Reply:
ok, if that can happen wouldn't it make more sense to say that "already exists" is not an issue? after all, we get the desired state we wish for

Reply:
I think the error msg is a bit confusing, in this specific case the creation of the server failed since I used a same name for the provider that was existed earlier and when I deleted it the PV wasn't removed (the bug I attached) so the next time I've tried to create provider with the same name it failed because the PV for it cant be created (the name already in use) so we got back this message (its the error that was returned that displayed as part of the condition) so its not indicating that the deployment already exist, simply that it failed so in this case it the desire outcome. and again this wont we happening once the second bug is fixed only used it as an example for this flow.

Reply:
@bkhizgiy ok, so in the case you describe, the problem is not with the name of the provider but with the name of the PV that already exists - imagine that someone would have created the PV with that name (without a provider, just a PV with the name we generate according to the provider name), we would have got the same error
so in that case, we should have said "the provider already exists - great, let's try to add the PV" - and repeat this until we are able to
the problem in our code is that, unlike k8s, we don't always generate unique names - in our case, we shouldn't set the name of the PV in the ObjectMeta but the `GenerateName` so the provided string would be the prefix of a unique name

Reply:
for identification we can/should set labels - and then it's easier to query stuff